tests/test_avro_schema.py

```diff
@@ -2,16 +2,270 @@
+    """
+    Backward compatibility: A new schema is backward compatible if it can be used to read the data
+    written in the previous schema.
+    """
+    msg = "adding a field with default is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema2, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a field w/o default is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema1)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+    msg = "changing field name with alias is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema4, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "evolving a field type to a union is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema6, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a type from a union is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema1, schema6)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a new type in union is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema7, schema6)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a type from a union is NOT a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema6, schema7)
+    assert res != SchemaCompatibilityResult.compatible(), msg
+
+
+def test_schemaregistry_basic_backwards_transitive_compatibility():
+    """
+    Backward transitive compatibility: A new schema is backward compatible if it can be used to read the data
+    written in all previous schemas.
+    """
+    msg = "iteratively adding fields with defaults is a compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema8, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema8, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "adding a field with default is a backward compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema2, schema1)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a default is a compatible change, but not transitively"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
+
+    msg = "removing a default is not a transitively compatible change"
+    res = ReaderWriterCompatibilityChecker().get_compatibility(schema3, schema2)
+    assert res == SchemaCompatibilityResult.compatible(), msg
```

Comment:
(Not implying it's bad and should be changed,) are these the same thing?

Reply:
If it's this way in the SR tests, we'd better leave it as is.

Reply:
That is how it is defined in the original test suite: https://github.com/confluentinc/schema-registry/blob/v5.0.3/core/src/test/java/io/confluent/kafka/schemaregistry/avro/AvroCompatibilityTest.java#L118-L121

Reply:
Gotcha